web/interceptor/user_auth.go

```diff
@@ -11,36 +11,76 @@ import (
 	"github.com/quickfeed/quickfeed/web/auth"
 )
 
+type UserInterceptor struct {
+	tm     *auth.TokenManager
+	logger *zap.SugaredLogger
+}
+
+func NewUserInterceptor(logger *zap.SugaredLogger, tm *auth.TokenManager) *UserInterceptor {
+	return &UserInterceptor{
+		tm:     tm,
+		logger: logger,
+	}
+}
+
+func (u *UserInterceptor) WrapStreamingHandler(next connect.StreamingHandlerFunc) connect.StreamingHandlerFunc {
+	return connect.StreamingHandlerFunc(func(ctx context.Context, conn connect.StreamingHandlerConn) error {
+		cookie := conn.RequestHeader().Get(auth.Cookie)
```

Comment:
Most of this code is similar to `WrapUnary`. Can we make a shared function that can be called from both?